src/qps/interface/qps.c

```diff
@@ -889,8 +889,11 @@ PetscErrorCode QPSSetFromOptions(QPS qps)
   TRY( QPSSetTolerances(qps,rtol,atol,dtol,maxit) );
   TRY( PetscOptionsBool("-qps_auto_post_solve","QPSSolve automatically triggers PostSolve","QPSSetAutoPostSolve",qps->autoPostSolve,&qps->autoPostSolve,NULL) );
   flg = PETSC_FALSE;
-  TRY( PetscOptionsBool("-qps_monitor","Switches QPS monitor","QPSMonitorSet",flg,&flg,NULL) );
+  TRY( PetscOptionsBool("-qps_monitor","Turn on default QPS monitor","QPSMonitorSet",flg,&flg,NULL) );
   if (flg) TRY( QPSMonitorSet(qps,QPSMonitorDefault,NULL,NULL) );
+  flg = PETSC_FALSE;
+  TRY( PetscOptionsBool("-qps_monitor_cancel","Turn off all QPS monitors","QPSMonitorCancel",flg,&flg,NULL) );
+  if (flg) TRY( QPSMonitorCancel(qps) );
```

Comment:
```suggestion
  TRY( PetscOptionsBool("-qps_monitor_cancel","Turn off all QPS monitors","QPSMonitorCancel",flg,&flg,NULL) );
  if (flg) TRY( QPSMonitorCancel(qps) );
  flg = PETSC_FALSE;
  TRY( PetscOptionsBool("-qps_monitor","Turn on default QPS monitor","QPSMonitorSet",flg,&flg,NULL) );
  if (flg) TRY( QPSMonitorSet(qps,QPSMonitorDefault,NULL,NULL) );
```
Cancel hardwired monitors but allow a user to set monitor from options. (E.g., I set a custom monitor routine in the code and I do want to turn that off and run with the default one.)